core/domain/question_domain.py

```diff
@@ -232,28 +252,40 @@ def update_language_code(self, language_code):
         """
         self.language_code = language_code
 
-    def update_question_data(self, question_data):
+    def update_question_state_data(self, question_state_data):
         """Updates the question data of the question.
 
         Args:
-            question_data: dict. A dict representing the question data.
+            question_state_data: dict. A dict representing the question state
```

Comment:
This seems wrong. It should be a State object.

If it really needs to be a dict then call it question_state_data_dict.

Reply:
The object received by the function here would be the value received from the frontend, so I think it would be a dict. Yeah, will rename it.

Reply:
Done.